unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -43,12 +43,13 @@
    * Finally is the unit name, ends with an optional "s".
    */
   private static String unitRegex(String unit) {
-    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?)?";
+    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?\\s*)?";
```

Comment:
@srowen  don't hang up on the package name for now. It's just a place to put all the data types implemented in Java. We will most likely rename the package.